File: backend/app/schemas/cameras.py
"""
Pydantic schemas for camera API endpoints.
"""
from datetime import datetime
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel, Field


class CameraListItem(BaseModel):
    """Camera item in list response."""
    id: UUID
    label: str
    client_id: UUID
    client_name: Optional[str] = None
    site_id: UUID
    site_name: Optional[str] = None
    manufacturer: Optional[str] = None
    model: Optional[str] = None
    status: str
    camera_protocol: Optional[str] = None
    camera_channel_number: Optional[int] = None
    parent_nvr_id: Optional[str] = None
    parent_nvr_label: Optional[str] = None
    health_status: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class CameraListResponse(BaseModel):
    """Paginated camera list."""
    items: List[CameraListItem]
    total: int
    page: int
    page_size: int


class CameraInfoResponse(BaseModel):
    """Detailed camera info with parent NVR."""
    id: UUID
    label: str
    client_id: UUID
    client_name: Optional[str] = None
    site_id: UUID
    site_name: Optional[str] = None
    manufacturer: Optional[str] = None
    model: Optional[str] = None
    serial_number: Optional[str] = None
    status: str
    camera_protocol: Optional[str] = None
    camera_channel_number: Optional[int] = None
    camera_stream_type: Optional[str] = None
    camera_rtsp_url_masked: Optional[str] = None
    parent_nvr_id: Optional[str] = None
    parent_nvr_label: Optional[str] = None
    parent_nvr_health_status: Optional[str] = None
    health_status: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/api/cameras.ts
import { apiClient } from './client';

export interface CameraListItem {
  id: string;
  label: string;
  client_id: string;
  client_name?: string;
  site_id: string;
  site_name?: string;
  manufacturer?: string;
  model?: string;
  status: string;
  camera_protocol?: string;
  camera_channel_number?: number;
  parent_nvr_id?: string;
  parent_nvr_label?: string;
  health_status?: string;
  created_at: string;
  updated_at: string;
}

export interface CameraListResponse {
  items: CameraListItem[];
  total: number;
  page: number;
  page_size: number;
}

export interface CameraInfoResponse {
  id: string;
  label: string;
  client_id: string;
  client_name?: string;
  site_id: string;
  site_name?: string;
  manufacturer?: string;
  model?: string;
  serial_number?: string;
  status: string;
  camera_protocol?: string;
  camera_channel_number?: number;
  camera_stream_type?: string;
  camera_rtsp_url_masked?: string;
  parent_nvr_id?: string;
  parent_nvr_label?: string;
  parent_nvr_health_status?: string;
  health_status?: string;
  created_at: string;
  updated_at: string;
}

export const camerasApi = {
  listCameras: async (params?: {
    client_id?: string;
    site_id?: string;
    page?: number;
    page_size?: number;
  }): Promise<CameraListResponse> => {
    const cleanParams = Object.fromEntries(
      Object.entries(params || {}).filter(([, v]) => v !== undefined && v !== '')
    );
    const response = await apiClient.get<CameraListResponse>('/cameras/', { params: cleanParams });
    return response.data;
  },

  getCameraInfo: async (assetId: string): Promise<CameraInfoResponse> => {
    const response = await apiClient.get<CameraInfoResponse>(`/cameras/${assetId}/info`);
    return response.data;
  },

  getSnapshot: async (assetId: string): Promise<Blob> => {
    const response = await apiClient.get(`/cameras/${assetId}/snapshot`, {
      responseType: 'blob',
    });
    return response.data;
  },
};
